src/huggingface_hub/utils/_dotenv.py

```diff
@@ -33,19 +33,23 @@ def load_dotenv(dotenv_str: str) -> Dict[str, str]:
             continue  # Skip comments and empty lines
 
         match = line_pattern.match(line)
-        if not match:
-            continue  # Skip malformed lines
-
-        key, raw_val = match.group(1), match.group(2) or ""
-        val = raw_val.strip()
-
-        # Remove surrounding quotes if quoted
-        if (val.startswith('"') and val.endswith('"')) or (val.startswith("'") and val.endswith("'")):
-            val = val[1:-1]
-            val = val.replace(r"\n", "\n").replace(r"\t", "\t").replace(r"\"", '"').replace(r"\\", "\\")
-            if raw_val.startswith('"'):
-                val = val.replace(r"\$", "$")  # only in double quotes
-
-        env[key] = val
+        if match:
+            key = match.group(1)
+            if match.group(2):  # if there is '='
+                raw_val = match.group(3) or ""
+                val = raw_val.strip()
+                # Remove surrounding quotes if quoted
+                if (val.startswith('"') and val.endswith('"')) or (val.startswith("'") and val.endswith("'")):
+                    val = val[1:-1]
+                    val = val.replace(r"\n", "\n").replace(r"\t", "\t").replace(r"\"", '"').replace(r"\\", "\\")
+                    if raw_val.startswith('"'):
+                        val = val.replace(r"\$", "$")  # only in double quotes
+            elif environ is not None and key in environ:
+                # Get it from the current environment
+                val = environ[key]
+            else:
+                continue
+
+            env[key] = val
```

Comment:
```suggestion
            val = None
            if match.group(2):  # if there is '='
                raw_val = match.group(3) or ""
                val = raw_val.strip()
                # Remove surrounding quotes if quoted
                if (val.startswith('"') and val.endswith('"')) or (val.startswith("'") and val.endswith("'")):
                    val = val[1:-1]
                    val = val.replace(r"\n", "\n").replace(r"\t", "\t").replace(r"\"", '"').replace(r"\\", "\\")
                    if raw_val.startswith('"'):
                        val = val.replace(r"\$", "$")  # only in double quotes
            elif environ is not None:
                # Get it from the current environment
                val = environ.get(key)

            if val is not None:
                env[key] = val
```

(slight preference for this which guarantees `val` is always defined, without looking at the internal code of each `if elif ` statement